app/components/charts-generic/axis/axis-width-time.tsx

```diff
@@ -27,28 +25,23 @@ export const AxisTime = () => {
 
   const hasNegativeValues = yScale.domain()[0] < 0;
 
-  const labelLengths = xUniqueValues.map((v) =>
-    estimateTextWidth(formatDateAuto(v))
-  );
-  const maxLabelLength = max(labelLengths, (d) => d) || 70;
+  // Approximate the longest date format we're using for
+  // Roughly equivalent to estimateTextWidth("99.99.9999", 12);
+  const maxLabelLength = 70;
 
   // This could be useful: use data points as tick values,
   // but it does not solve the problem of overlapping.
   // const tickValues =
   //   bounds.chartWidth / (maxLabelLength + 20) > xUniqueValues.length
   //     ? xUniqueValues
   //     : null;
-  const ticks = Math.min(
```

Comment:
This comparison is useful to prevent time labels for non existing data points, like months when there are only yearly values.

Reply:
I found that it's not a very useful heuristic, since time is continuous and not ordinal, so the axis should offer a reference and not the distinct values. So I personally don't mind seeing months between the year ticks. After all, we show ticks at regular intervals other continuous axes too!

Also it gets more tricky: Let's say we have only two data points, at e.g. 2017-01-08 and 2018-01-12. If we limit the tick number to 2, the axis will only show one at 2008! We could argue to show the exact start and end but I think showing month intervals is more readable than using the same precision like the data.

Before (using the FOAG dataset):

<img width="694" alt="Bildschirmfoto 2020-05-28 um 15 44 20" src="https://user-images.githubusercontent.com/70969/83149226-26b6be00-a0fa-11ea-9ea6-fa17deb3ac52.png">

After:

<img width="685" alt="Bildschirmfoto 2020-05-28 um 15 44 12" src="https://user-images.githubusercontent.com/70969/83149249-2cac9f00-a0fa-11ea-8277-4070de588785.png">

It may not be the _perfect_ way to label this axis but I think it's better to have more ticks.

Reply:
The analogy with a continuous quantitative axis is very convincing! You're right, let's have additional ticks to serve as a time reference. Maybe @janwaechter also has an opinion on this?

Reply:
Interesting idea. I also agree that it looks better, but I also fear that it might lead people to assume that there are more data-points visible then there actually are, no?

Honestly, I guess in this case they shouldn't even use a line-chart, but rather a column chart. 😄

Reply:
@janwaechter Yeah, the problem is that we can't switch the chart type easily (keeping the filters and settings) at the moment ;) Some filters obviously have more data points and some have none or just one (which also doesn't get displayed in a line chart!).

I firmly disagree though that ticks on a time axis should align with the data points. I think this would make the axis super confusing and too precise (and there would be no way to leave ticks out).

A solution to indicate _where_ data points are could be to display dots on top of the lines (like we do on hover). This would also address the "single data point" problem, at least it's not completely unusable.

Reply:
@herrstucki True that, you're absolutely right — the problem are not the ticks themselves, but rather the uncertainty about where the data points are on said line…adding point-markers could help. Not sure if I like them from a «visual» perspective, though. 😅